### PR TITLE
feat: smarter intra-session set adjustment (#158)

### LINKED ISSUE #158: enhance(workout): smarter intra-session set adjustment beyond weight-only RIR lookup
### Goal
Implement a 'Triple Progression' system (Reps x Weight x Sets) that adapts intra-session coaching to the specific mechanics of the exercise (reps-based vs. bodyweight/cable-based).

### Why: The Foundation of Strength
Progressive Overload is the fundamental law of muscular adaptation. Most fitness applications act as passive logs—they record what was done but do not guide the user on how to improve. By automating the "Triple Progression" logic (autoregulation), we turn the application into a 'smart coach' that adapts to daily performance and equipment limitations, ensuring every session pushes the user closer to their adaptation threshold safely and consistently.

### Core Architecture
- **Progression Hierarchy:** Volume (Reps) -> Intensity (Weight) -> Density (Sets).
- **Equipment Tiering:**
    - **Weight-first:** (barbell, dumbbell, machine) → Prioritizes weight increments.
    - **Reps-only:** (bodyweight, cable, band) → Weight changes are suppressed/ignored; progression happens through reps and sets.
- **Auto-Loading:** If a bodyweight exercise is loaded (user logs weight > 0), the engine treats it as 'Weight-first' for the remainder of the session.

### Progression Logic
- **Shortfall Threshold:** `range_min` (Hitting below this = HOLD/DELOAD).
- **Progression Trigger:** `range_max` (Hitting this at RIR 4+ = SUGGEST PROGRESSION).
- **Output:** Always a concrete integer (e.g., "10"), never a range.

### Technical Tasks
- [ ] Refactor `computeIntraSessionSuggestion` to accept targets and history.
- [ ] Implement hierarchy logic: REPS_UP -> WEIGHT_UP -> SETS_UP.
- [ ] Add equipment-aware logic to suppress weight suggestions where invalid.
- [ ] Update `SetsTable` to consume and apply progression targets.
- [ ] Unit test all 8+ edge cases (failure, progression, ceiling reached).

### Acceptance Criteria
- Engine provides recommendations for reps, duration, and weighted bodyweight consistently.
- No UI logic duplication for duration-based progression.
- Progression logic is unified and easy to extend.

### PR DESCRIPTION
**Pull request by PierreTsia**
## What

- Rewrite the intra-session coaching engine (`rirSuggestion.ts`) with fatigue detection, equipment-aware adjustment tiers (weight-first / reps-only), a full coaching rule table, and cascade batch updates with a 2-step deload cap
- Wire the new engine into `SetsTable.confirmRir` — replacing the single-next-set update with a cascade that updates **all** remaining eligible sets in one pass
- Add `manuallyEdited` flag to `SessionSetRow` types so user overrides are respected by the batch update
- Remove the legacy shim, clean up imports

## Why

The previous intra-session logic was a simple 3-bucket RIR check that only touched the next set. It didn't account for fatigue trends, rep shortfalls, equipment differences (bodyweight vs barbell vs dumbbell increments), or the fact that users manually edit future sets. This made the coaching feel dumb after set 2 and led to frequent manual overrides — exactly what the app should prevent.

## How

- **T49 (engine):** New types (`AdjustmentTier`, `CompletedSetInfo`, `RepRange`, `IntraSessionContext`), helpers (`getAdjustmentTier`, `parseTargetRepRange`, `detectFatigue`), two rule tables (weight-first / reps-only with proportional deload), and `computeCascadeSuggestions` with tier-stable 2-step deload cap. 69 unit tests.
- **T50 (integration):** `SetsTable.updateField` sets `manuallyEdited: true` on actual changes. `SetsTable.confirmRir` collects completed sets → parses target rep range → filters eligible remaining sets (skipping done, duration, and manually-edited) → calls cascade → applies suggestions via a `Map` lookup in a single `.map()`. 7 new integration tests covering batch update, manual override, cascade cap, reps-only, AMRAP passthrough, duration immunity.
- **Functional style:** Cascade application uses declarative `.filter().map()` pipelines with no mutable accumulators — codified in a new cursor rule.

Closes #158

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: src/lib/rirSuggestion.ts [hunk 1]
```diff
@@ -1,29 +1,288 @@
...
-  if (prevRir === 0) {
-    if (prevWeight <= 0) return { weight: 0, reps: prevReps }
-    return { weight: Math.max(inc, prevWeight - inc), reps: prevReps }
+  switch (bucket) {
+    case "failure": {
+      const newWeight = w <= 0 ? 0 : Math.max(inc, w - inc)
+      return {
+        weight: newWeight,
+        reps: shortfall ? String(reps) : String(reps),
+      }
```
[the remark this answers is not in the document]
Comment (PierreTsia) [reply src/lib/rirSuggestion.ts]: Good catch — `Math.max(inc, w - inc)` could indeed push weight *up* when `w < inc`.

Fixed: the deload now uses `Math.min(w, Math.max(inc, w - inc))` so the suggestion never exceeds the current weight.

Addressed in `71ddcad` (71ddcad31edf2fe0e35e5409d4b54be91d690cfa)

File: src/components/workout/SetsTable.tsx [hunk 2]
```diff
@@ -355,28 +363,51 @@ export function SetsTable({
...
+
+      const suggestions =
+        eligible.length > 0 && completedSets.length > 0
+          ? computeCascadeSuggestions(
+              completedSets,
+              eligible.length,
+              targetRepRange,
+              unit,
+              equipment ?? "",
+            )
```
[the remark this answers is not in the document]
Comment (PierreTsia) [reply src/components/workout/SetsTable.tsx]: Same as above — intentional v1 simplification, follow-up in #163 to thread `weight_increment` through the engine.
